frontend/src/App.js

```diff
@@ -40,34 +68,76 @@ FROM ( SELECT MONTH(committer_when) as month,
 
     this.showCode = this.showCode.bind(this);
     this.showUAST = this.showUAST.bind(this);
+  }
+
+  setState(partialState, callback) {
+    super.setState(partialState, callback);
 
-    this.uniqueKey = 0;
+    if (typeof partialState.results !== 'undefined') {
+      persist('results', partialState.results);
+    }
+    if (typeof partialState.history !== 'undefined') {
+      persist('history', partialState.history);
+    }
   }
 
   handleTextChange(text) {
     this.setState({ sql: text });
   }
 
   setResult(key, result) {
-    if (!this.state.results.has(key)) {
+    const { results, history } = this.state;
+    const historyIdx = history.findIndex(i => i.key === key);
+
+    if (historyIdx >= 0) {
+      const status =
+        typeof result.response !== 'undefined' ? STATUS_SUCCESS : STATUS_ERROR;
+      const newHistory = [
+        ...history.slice(0, historyIdx),
+        {
+          ...history[historyIdx],
+          status,
+          errorMsg: result.errorMsg
+        },
```

Comment:
Could you please help me understand, if a tab title gets persisted in the history?